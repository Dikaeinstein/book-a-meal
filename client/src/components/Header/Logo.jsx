import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';

const Logo = ({ text, className }) => (
  <NavLink
    to="/"
    activeClassName="active"
  >
    <h1 className={className}>{text}</h1>
  </NavLink>
);

Logo.propTypes = {
  text: PropTypes.string.isRequired,
  className: PropTypes.string.isRequired,
};

export default Logo;
