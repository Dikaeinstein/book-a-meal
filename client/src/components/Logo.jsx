import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import { userSignOut } from '../actions/userActions';

const ConnectedLogo = ({ text, className, goHome }) => {
  const handleClick = () => {
    goHome();
  };
  return (
    <NavLink to="/" onClick={handleClick}>
      <h1 className={className}>{text}</h1>
    </NavLink>
  );
};

ConnectedLogo.propTypes = {
  text: PropTypes.string.isRequired,
  className: PropTypes.string.isRequired,
  goHome: PropTypes.func.isRequired,
};

const mapDispatchToProps = dispatch => ({
  goHome: () => dispatch(userSignOut()),
});

const Logo = connect(null, mapDispatchToProps)(ConnectedLogo);

export default Logo;
