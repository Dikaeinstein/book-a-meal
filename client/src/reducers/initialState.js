const initialState = {
  urls: [
    {
      id: 1,
      name: 'Signin',
      link: 'signin',
    },
    {
      id: 2,
      name: 'Signout',
      link: 'signout',
    },
  ],
  menus: {
    isFetching: false,
    data: {
      menu: {
        name: 'Menu for today',
        meals: [],
      },
      menus: [],
    },
    isSet: false,
    isSaving: false,
    fetchError: null,
    saveError: null,
    updateError: null,
    isUpdating: false,
  },
  user: {
    isSubmitting: false,
    loggedIn: false,
    data: {
      id: null,
      name: '',
      email: '',
      role: '',
    },
    error: null,
  },
  orders: {
    isFetching: false,
    isSaving: false,
    isUpdating: false,
    isDeleting: false,
    data: {
      userOrders: [],
      checkedOutOrder: {},
    },
    fetchError: null,
    saveError: null,
    updateError: null,
    deleteError: null,
  },
  meals: {
    isFetching: false,
    isUpdating: false,
    isSaving: false,
    data: [],
    fetchError: null,
    saveError: null,
    updateError: null,
  },
  dashboard: {
    isFetchingTotalOrders: false,
    isFetchingTotalSales: false,
    isUpdating: false,
    isDeleting: false,
    fetchTotalOrdersError: null,
    fetchTotalSalesError: null,
    isFetchingAllOrders: false,
    fetchAllOrdersError: null,
    deleteOrderError: null,
    updateOrderError: null,
    totalAmount: 0,
    totalOrders: 0,
    allOrders: [],
  },
};

export default initialState;
