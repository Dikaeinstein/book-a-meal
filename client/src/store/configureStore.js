import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import { createLogger } from 'redux-logger';
import throttle from 'lodash.throttle';
import { loadState, saveState } from '../helpers/localStorage';
import initialState from '../reducers/initialState';
import rootReducer from '../reducers';

/**
 * Return redux middleware
 */
const getMiddleware = () => {
  const logger = createLogger();

  const middleware =
    process.env.NODE_ENV !== 'production'
    && process.env.NODE_ENV !== 'test'
      ? applyMiddleware(thunk, logger)
      : applyMiddleware(thunk);

  return middleware;
};

/**
 * Configure the redux store
 *
 * @returns {object} Redux store
 */
const configureStore = () => {
  const persistedState = loadState();

  const state = persistedState
    ? { ...initialState, ...persistedState }
    : initialState;

  const store = createStore(
    rootReducer,
    state,
    compose(
      getMiddleware(),
      window.devToolsExtension ? window.devToolsExtension() : empty => empty,
    ),
  );

  // Throttle the saveState function to be called once every second
  store.subscribe(throttle(() => {
    saveState({ orders: store.getState().orders });
  }, 1000));

  return store;
};

export default configureStore;
