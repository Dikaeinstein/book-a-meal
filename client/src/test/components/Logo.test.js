import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { NavLink, MemoryRouter } from 'react-router-dom';
import Logo from '../../components/Header/Logo';
import configureStore from '../../store/configureStore';
import initialState from '../../reducers/initialState';
import setup from '../setup';

// Setup tests
setup();

let wrapper;

const store = configureStore(initialState);

describe('<Logo />', () => {
  beforeAll(() => {
    /* eslint function-paren-newline: 0 */
    wrapper = mount(
      <MemoryRouter>
        <Logo
          text="Book-A-Meal"
          className="col-1-4 logo text-left"
          store={store}
        />
      </MemoryRouter>,
    );
  });
  it('should render a NavLink', () => {
    expect(wrapper.find(NavLink).length).toEqual(1);
  });
  it('should render a h1 element with correct text', () => {
    expect(wrapper.find('h1').text()).toEqual('Book-A-Meal');
  });
  it('should match snapshot', () => {
    const tree = mount(
      <MemoryRouter>
        <Logo
          text="Book-A-Meal"
          className="col-1-4 logo text-left"
          store={store}
        />
      </MemoryRouter>,
    );
    expect(toJson(tree)).toMatchSnapshot();
  });
});
