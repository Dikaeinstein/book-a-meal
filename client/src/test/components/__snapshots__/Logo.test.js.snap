// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo /> should match snapshot 1`] = `
<MemoryRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "oidat6",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "oidat6",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Connect(ConnectedLogo)
      className="col-1-4 logo text-left"
      store={
        Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        }
      }
      text="Book-A-Meal"
    >
      <ConnectedLogo
        className="col-1-4 logo text-left"
        goHome={[Function]}
        store={
          Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          }
        }
        text="Book-A-Meal"
      >
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          onClick={[Function]}
          to="/"
        >
          <Route
            path="/"
          >
            <Link
              aria-current="true"
              className="active"
              onClick={[Function]}
              replace={false}
              style={Object {}}
              to="/"
            >
              <a
                aria-current="true"
                className="active"
                href="/"
                onClick={[Function]}
                style={Object {}}
              >
                <h1
                  className="col-1-4 logo text-left"
                >
                  Book-A-Meal
                </h1>
              </a>
            </Link>
          </Route>
        </NavLink>
      </ConnectedLogo>
    </Connect(ConnectedLogo)>
  </Router>
</MemoryRouter>
`;
